### ISSUE TITLE
Can´t paginate while using fence

### ISSUE THREAD

**Issue by Moyhdez** [malhal/Laravel-Geographical#5]
When i try to paginate using scopeFence i get this error:

(2/2) QueryException
SQLSTATE[42S22]: Column not found: 1054 Unknown column 'distance' in 'having clause' (SQL: select count(*) as aggregate from `spots` inner join `favorites` on `spots`.`id` = `favorites`.`spot_id` where `favorites`.`user_id` = 1 having distance BETWEEN 0 AND 150)

**Comment by YavorK** [comment]
Yes, same with the other method, but you can do it with the query builder...
DISCLAIMER: did not try it with the *fence* method but should be VERY similar.
```
public function index(Request $request)
  $requestPage = $request->input('page') ?? 1; // rewrite to $request->input('page') !== null ?  $request->input('page') : 1;  for older PHP versions
  $resultsPerPage = 30;
  $places = Place::distance($lat, $lng)
    ->orderBy('distance', 'ASC')
    ->skip(($requestPage - 1) * $resultsPerPage)
    ->take($resultsPerPage)
    ->get();
  return $places;
}
```

**Comment by Ruitjes** [comment]
You can also do $query = Ads::geofence($json[0]->lat, $json[0]->lon, 0, $distance)->get();
                
                $collection = collect($query);

                $page = 1;
                $perPage = 25;

                $ads = new LengthAwarePaginator($collection->forPage($page, $perPage), $collection->count(), $perPage, $page, ['path'=>url('api/products')]);`

**Comment by veneliniliev** [comment]
check this package https://github.com/justbetter/laravel-pagination-with-havings
